From c21f8562457b994b7e4ba46cba9fd05dc0e25e20 Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Thu, 28 Sep 2017 10:58:24 +0000
Subject: [PATCH] staging: fsl-dpaa2/mac: Remove redundant free

free_netdev (put_device) already handles freeing the private data
structure, and KASAN will complain due to a free after free if we
explicitly do the same afterwards.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 1 -
 1 file changed, 1 deletion(-)

--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -638,7 +638,6 @@ static int dpaa2_mac_remove(struct fsl_m
 	free_netdev(priv->netdev);
 
 	dev_set_drvdata(dev, NULL);
-	kfree(priv);
 
 	return 0;
 }
